### PR TITLE
ci(python-test): DS-2129 update python-test

### PR DESCRIPTION
**Pull request by mattfraz**
## SUMMARY
- Create python-default executor
- Create python-test command
- Create python-test-dsa job that uses python-test command
- Update to python-test job to use python-test command

Related PR: https://github.com/GoodwayGroup/ds-bidrange-model/pull/132

## TESTING
https://app.circleci.com/pipelines/github/GoodwayGroup/ds-bidrange-model/596/workflows/1f6be537-7ce9-4ed9-a674-00640750f95b/jobs/1063

## CRITICAL REVIEW AREAS
- CircleCI deployment
- Poetry/python config

### CODE REVIEW

File: src/commands/test-python.yml [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+              - store_artifacts:
+                  path: << parameters.tar-coverage-artifacts-path >>
+        - when:
+            condition: << parameters.tar-coverage-artifacts >>
+            steps:
+              - run:
+                  name: Compress Coverage Artifacts
+                  command: tar -cvzf coverage.tar << parameters.tar-coverage-artifacts-path >>
+              - store_artifacts:
+                  path: << parameters.tar-coverage-artifacts-path >>
```
Comment (bnormington) [inline src/commands/test-python.yml]: This is going to still upload every coverage file in the tar-coverage-artifacts-path. What we want is to upload just the coverage.tar file that was created from the previous step.  

```suggestion
                  path: coverage.tar
```


Also `tar-coverage-artifacts-path` param name should drop `tar` as it doesn't have that much to do with taring up the files. It's more the location of the coverage reports so something like `coverage-artifacts-path` or just `coverage-path`

File: src/executors/python-default.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+description: Common cimg/node image executor
```
Comment (bnormington) [inline src/executors/python-default.yml]:
```suggestion
description: Common cimg/python image executor
```

File: src/executors/python-default.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+parameters:
+  repo:
+    type: string
```
Comment (bnormington) [inline src/executors/python-default.yml]: I don't think we need a repo param but if we do then we should default to circle ci's default `cimg/pthon`
```suggestion
  repo:
    type: string
    default: cimg/python
```

File: src/executors/python-default.yml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+      - medium
+      - medium+
+      - large
+      - xlarge
+      - 2xlarge
+      - 2xlarge+
+    type: enum
+working_directory: ~/repo
+docker:
+  - image:  << parameters.repo >>/python:<< parameters.tag >>
```
Comment (bnormington) [inline src/executors/python-default.yml]: If you make the change above then we should update this too
```suggestion
  - image:  << parameters.repo >>:<< parameters.tag >>
```

File: src/jobs/test-python.yml [hunk 5]
```diff
@@ -21,50 +29,46 @@ parameters:
...
+      If enabled, tars the ./coverage directory into coverage.tar for storage.
+  tar-coverage-artifacts-path:
+    type: string
+    default: ./coverage
+    description: |
+      Path of coverage directory
 
 executor: 
   name: dockerhub-authed
   repo: << parameters.executor-repo >>
```
Comment (bnormington) [inline src/jobs/test-python.yml]: I cant leave a comment on the `executor-rep` param on line 6 but that image `circleci/python` is old and is deprecated, we should go ahead and update it to `cimg/python`

File: src/executors/python-default.yml (hunk 3 above)
Comment (mattfraz) [reply src/executors/python-default.yml]: I can make these suggestions as well as update [source-inventory](https://github.com/GoodwayGroup/source-inventory/blob/3ffd9f87621a7f12b8bdbadd3ff9d152b6bec519/.circleci/config.yml#L11) which is the only repo that overrides the `executor-repo` to use the current python image.

**Review by bnormington — APPROVED** [review]
Thanks for the updates

**Pull request merged**